rsky-relay/src/server/server.rs

```diff
@@ -262,6 +304,64 @@ impl Server {
         }
     }
 
+    #[cfg(not(feature = "labeler"))]
+    fn list_hosts(&mut self, url: &Url) -> Result<ListHosts> {
+        // Default query parameters.
+        let mut limit = 200;
+        let mut cursor = None;
+
+        for (key, value) in url.query_pairs() {
+            match key.as_ref() {
+                "limit" => match value.parse::<u16>() {
+                    Ok(l @ 1..=1000) => limit = l,
+                    _ => {
+                        return Err(eyre!("limit parameter invalid or out of range: {value}"));
+                    }
+                },
+                "cursor" => match value.parse::<i64>() {
+                    Ok(c) => cursor = Some(c),
+                    Err(_) => {
+                        return Err(eyre!("cursor parameter invalid: {value}"));
+                    }
+                },
+                // Ignore unknown query parameters.
+                _ => (),
+            }
+        }
+
+        let mut stmt_hosts = self.relay_conn.prepare_cached(
+            "SELECT rowid, host, cursor
+            FROM hosts
+            WHERE :cursor is NULL OR rowid > :cursor
+            LIMIT :limit;",
+        )?;
+        let hosts = stmt_hosts
+            .query_map(
+                named_params! {
+                    ":cursor": cursor,
+                    ":limit": limit,
+                },
+                |row| Ok((row.get::<_, i64>("rowid")?, row.get("host")?, row.get("cursor")?)),
+            )?
+            .collect::<Result<Vec<_>, _>>()?;
+
+        let cursor = hosts.last().map(|(rowid, _, _)| rowid.to_string());
+
+        let hosts = hosts
+            .into_iter()
+            .map(|(_, hostname, seq)| Host {
+                // TODO: Track host account counts.
+                account_count: 0,
+                hostname,
+                seq,
+                // TODO: Track status of hosts.
+                status: HostStatus::Active,
```

Comment:
indigo does things like [mark a host as offline if it fails repeated connection attempts](https://github.com/bluesky-social/indigo/blob/c5eaa30f683f959af20ea17fdf390d8a22d471cd/cmd/relay/relay/slurper.go#L334), or [mark hosts that are actually relays as banned](https://github.com/bluesky-social/indigo/blob/c5eaa30f683f959af20ea17fdf390d8a22d471cd/cmd/relay/relay/slurper.go#L343-L351). I couldn't find any logic in rsky-relay that did this, so for now this reports [the same default value as indigo](https://github.com/bluesky-social/indigo/blob/c5eaa30f683f959af20ea17fdf390d8a22d471cd/cmd/relay/relay/models/models.go#L44). This does mean rsky-relay is over-reporting which hosts it is actively connected to, but that can be fixed by someone more familiar with the stack.